crates/biome_graphql_syntax/src/file_source.rs

```diff
@@ -0,0 +1,68 @@
+use std::ffi::OsStr;
+use std::path::Path;
+use biome_rowan::FileSourceError;
+
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[derive(
+    Debug, Clone, Default, Copy, Eq, PartialEq, Hash, serde::Serialize, serde::Deserialize,
+)]
+pub struct GraphqlFileSource {}
+
+impl GraphqlFileSource {
+
+    /// Try to return the CSS file source corresponding to this file name from well-known files
+    pub fn try_from_well_known(file_name: &str) -> Result<Self, FileSourceError> {
+        // TODO: to be implemented
+        Err(FileSourceError::UnknownFileName(file_name.into()))
+    }
+    
+    /// Try to return the CSS file source corresponding to this file extension
+    pub fn try_from_extension(extension: &str) -> Result<Self, FileSourceError> {
+        match extension {
+            "graphql" => Ok(Self::default()),
```

Comment:
```suggestion
            "graphql" | "gql" => Ok(Self::default()),
```

https://github.com/graphql/graphiql/tree/main/packages/vscode-graphql#graphql-gql-file-extension-support-and-gqlgraphql-tagged-template-literal-support-for-tsx-jsx-ts-js

Reply:
I didn't know that :O